File: Domain/Models/Vehiculos/Vehiculo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Domain.Models.Finanzas;
using Domain.Models.Personas;

namespace Domain.Models.Vehiculos
{
    public class Vehiculo : ClaseBase
    {
        [Key]
        public int VehiculoId { get; set; }
        
        public string Dominio { get; set; }
        public Procedencia? Procedencia { get; set; }
        public DateTime FechaInscripcionInical { get; set; }
        public Marca Marca { get; set; }
        public Modelo Modelo { get; set; }
        public TipoVehiculo? TipoVehiculo { get; set; }

        [Column(TypeName = "tinyint")]
        public Int16 Ano { get; set; }

        [Column(TypeName = "nvarchar(20)")]
        public string NumeroMotor { get; set; }

        [Column(TypeName = "nvarchar(20)")]
        public string NumeroChasis { get; set; }

        [Column(TypeName = "nvarchar(20)")]
        public string MarcaMotor { get; set; }

        [Column(TypeName = "nvarchar(20)")]
        public string MarcaChasis { get; set; }

        public DateTime FechaAdquisicion { get; set; }

        public RevisionTecnica? RevisionTecnica { get; set; } 

        public List<Transaccion>? Transacciones { get; set; }

        public Titular? Titular { get; set; } 



    }
}

File: CarAgency/ClientApp/src/domain/models/vehiculos/Vehiculo.ts
import {IVehiculo} from "domain/interfaces/vehiculos/IVehiculo";
import {Transaccion} from "domain/models/finanzas/Transaccion";
import {Marca} from "domain/models/vehiculos/Marca";
import {Modelo} from "domain/models/vehiculos/Modelo";
import {TipoVehiculo} from "domain/models/vehiculos/TipoVehiculo";
import { Titular } from "domain/models/personas/Titular";
import Procedencia from "domain/models/vehiculos/Procedencia";

import RevisionTecnica from "domain/models/vehiculos/revisionestecnicas/RevisionTecnica";

export class Vehiculo implements IVehiculo{
    constructor(){
      this.VehiculoId = undefined;
      this.Dominio = "";
      this.Procedencia= new Procedencia();
      this.FechaInscripcionInical= new Date();
      this.Marca = new Marca();
      this.Modelo= new Modelo();
      this.Ano= 2020;
      this.NumeroMotor = "";
      this.NumeroChasis= "";
      this.MarcaMotor= "";
      this.MarcaChasis= "";
      this.FechaAdquisicion= new Date();
      this.RevisionTecnica = undefined;
      this.Transacciones = undefined; 
      this.Titular = new Titular();
    };
  
    VehiculoId?: number;
    Dominio: string;
    Procedencia: Procedencia;
    FechaInscripcionInical: Date;
    Marca: Marca;
    Modelo: Modelo;
    Ano: number;
    NumeroMotor: string;
    NumeroChasis: string;
    MarcaMotor: string;
    MarcaChasis: string;
    FechaAdquisicion: Date;
    RevisionTecnica?: RevisionTecnica;
    Transacciones?: Transaccion[];
    Titular: Titular;
  }